test/mk/compiler.mk

```diff
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: Apache-2.0 OR ISC OR MIT
+#
+# Compiler feature detection for mlkem-native
+# This file detects whether the compiler supports various architectural features
+# used by mlkem-native through compile-time tests with C code containing inline assembly.
+#
+# Feature detection can be overridden by setting the corresponding variable on the command line:
+#   make MK_COMPILER_SUPPORTS_SHA3=0    # Disable SHA3 detection
+#   make MK_COMPILER_SUPPORTS_AVX2=0    # Disable AVX2 detection
+#   make MK_COMPILER_SUPPORTS_BMI2=0    # Disable BMI2 detection
+#   make MK_COMPILER_SUPPORTS_SSE2=0    # Disable SSE2 detection
```

Comment:
Do we want to prefix those with the architecture? It is not unimaginable that X86 introduces SHA3 instructions.

Reply:
Okay.